samcli/lib/iac/interface.py

```diff
@@ -739,6 +739,12 @@ def stacks(self) -> List[Stack]:
     def stacks(self, stacks: List[Stack]) -> None:
         self._stacks = stacks
 
+    @property
+    def default_stack(self) -> Optional[Stack]:
+        if len(self.stacks) > 0:
```

Comment:
Is there a reason for checking the length of `self.stacks` but returning a stack from `self._stacks`?

Reply:
It's to ensure `self.stacks` contains at least one stack. Otherwise, it returns `None`.

Reply:
I just meant why the check is for the public member `self.stacks` but the return is a value of the protected member `self._stacks`. Just for my understanding.

Reply:
ah got it. They are the same but I agree that it should be either both internal or both public. I'll update it

Reply:
updated